packages/core/src/utils.ts

```diff
@@ -72,6 +72,54 @@ export function throttle(
   }
 }
 
+/**
+ * Performs a deep merge of objects and arrays
+ * - arrays values are merged index by index
+ * - objects are merged by keys
+ * - values get replaced, unless undefined
+ *
+ * ⚠️ this method does not prevent infinite loops while merging circular references ⚠️
+ *
+ * @param {destination, ...toMerge} any - values to merge
+ * @returns {any} New merged value
+ */
+const isContextArray = (value: ContextValue): value is ContextArray => Array.isArray(value)
+const isContext = (value: ContextValue): value is Context => !Array.isArray(value) && typeof value === 'object'
+
+export function deepMerge(destination: ContextValue, ...toMerge: ContextValue[]): ContextValue {
```

Comment:
Since we were relying on lodash, I am not sure that we have test cases that ensure the behavior of this function.
It would be nice to have a set of test cases to validate it and ease future maintenance.

Reply:
I'll add unit tests indeed